src/ShallowTraversal.js

```diff
@@ -95,6 +96,10 @@ export function nodeHasType(node, type) {
   return node.type.name === type || node.type.displayName === type;
 }
 
+export function nodeHasProps(node, props) {
```

Comment:
This function name clashes quite a lot with the `nodeHasProperty` function. I think it should probably be more explicit so it's known that it's meant for the object selector.

Something like `nodeMatchesObjectProps` ?